packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,274 @@
+import { InstanceType, ISecurityGroup, SubnetSelection } from '@aws-cdk/aws-ec2';
+import { IRole, ManagedPolicy, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Construct, IResource, Resource } from '@aws-cdk/core';
+import { Cluster } from './cluster';
+import { CfnNodegroup } from './eks.generated';
+
+/**
+ * NodeGroup interface
+ */
+export interface INodegroup extends IResource {
+  /**
+   * Name of the nodegroup
+   * @attribute
+   */
+  readonly nodegroupName: string
+}
+
+/**
+ * The AMI type for your node group. GPU instance types should use the `AL2_x86_64_GPU` AMI type, which uses the Amazon EKS-optimized
+ * Linux AMI with GPU support. Non-GPU instances should use the `AL2_x86_64` AMI type, which uses the Amazon EKS-optimized Linux AMI.
+ */
+export enum NodegroupAmiType {
+  /**
+   * Amazon EKS-optimized Linux AMI
+   */
+  AL2_X86_64 = 'AL2_x86_64',
+  /**
+   *  Amazon EKS-optimized Linux AMI with GPU support
+   */
+  AL2_X86_64_GPU = 'AL2_x86_64_GPU',
+}
+
+/**
+ * The remote access (SSH) configuration to use with your node group.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-eks-nodegroup-remoteaccess.html
+ */
+export interface NodegroupRemoteAccess {
+  /**
+   * The Amazon EC2 SSH key that provides access for SSH communication with the worker nodes in the managed node group.
+   */
+  readonly ec2SshKey: string;
```

Comment:
rename to sshKeyName (otherwise one might think this is an actual key), and I believe the ec2 terminology is simply “ssh key name”.